### ISSUE TITLE
Option to enable verbatim mode by default

### ISSUE THREAD

**Issue by TobyBrull** [doxygen/doxygen#8399]
I use doxygen only to generate XML output and then pass the XML output to Sphinx (via Breathe; basically as explained [here](https://devblogs.microsoft.com/cppblog/clear-functional-c-documentation-with-sphinx-breathe-doxygen-cmake/)). Since Sphinx is build around reStructuredText, I also want to exclusively write reStructuredText in my C++ special comment blocks and not use Doxygen markup. To help with that, I've added the following aliases to my `Doxyfile`.

```
ALIASES  = "rst=\verbatim embed:rst:leading-asterisk"
ALIASES += "endrst=\endverbatim"
```

In trivial cases one can, of course, simply not use any markup, but other than that now all my C++ special comment blocks look like the following.

```
/// \rst
///
/// Some **very**
/// important `documentation`.
///
/// \endrst 
```

This seems a bit redundant and I'd rather just like to write the following and have every special comment block be interpreted as reStructuredText.

```
/// Some **very**
/// important `documentation`.
```

Would it be possible to add options to Doxygen that would allow this? For example, one could extend the `Doxyfile` with the following two options that would be prepended/appended to every parsed special comment block:

```
SPECIAL_COMMENT_BLOCK_PREAMBLE = "\verbatim embed:rst:leading-asterisk"
SPECIAL_COMMENT_BLOCK_POSTAMBLE = "\endverbatim"
```

Maybe there are other, better alternatives?

Potentially related: https://github.com/doxygen/doxygen/issues/7791

**Comment by albert-github** [comment]
I assume with special comment blocks you mean blocks wit starting `///` or `/**` comments, but in that case what would happen with the following code:
```
/// a nice function
///
/// \param arg oops I have an argument
void fie(int arg);
```

So please  please attach a, small, self contained example (source+configuration file in a tar or zip) that shows better the wanted extension. Please don't add external links as they might not be persistent.

**Comment by TobyBrull** [comment]
Yes, by special comment blocks I meant blocks starting with `///` or `/**` (or other, as per the [Doxygen documentation](https://www.doxygen.nl/manual/docblocks.html#specialblock)).

Your code would be treated as if one had written the following without the proposed extension.
```
/// \verbatim embed:rst:leading-asterisk
/// a nice function
///
/// \param arg oops I have an argument
/// \endverbatim
void fie(int arg);
```
It wouldn't look nice. Basically, people who want to use Doxygen markup wouldn't be expected to use the extension.

Here is a small example that currently works without the proposed extension [issue_8399.zip](https://github.com/doxygen/doxygen/files/6054497/issue_8399.zip). The goal of the extension would be to get rid of the "\rst" and "\endrst" bits in `Example.hpp`. I could also amend that example with a little Linux Bash script that shows how the XML is used in Sphinx. Let me know if you would find that helpful.

**Comment by albert-github** [comment]
I, personally, don't think doing such a thing would be a good idea as it would block all doxygen commands and also some relations between elements.

I can think of some solutions:
- write an output driver outputting rest code (maybe xml code with rest c0mmands
- create, analogous to possibilities for markdown, a possibility to parse the command blocks as rest input. This will require a good, authoritative, description of the rest features.
- create a command (like the given `\rst ... \endrst` command block) that says that the input is to be treated as rest commands , so either:
  - needing a parsing like in the above
  - using the ALIAS as shown above but maybe with some extra features

**Comment by arwedus** [comment]
afaik the breathe extension only recognizes rST syntax inside verbatim sections ... and still parses sections like \\param and \\brief specially and separately. The use case to use rST as markup language inside C++ code Doxygen comments everywhere is thus also not supported by the sphinx extension `breathe`.

See also: https://breathe.readthedocs.io/en/latest/markups.html
